client/src/components/forms/submit-button.component.test.tsx

```diff
@@ -0,0 +1,37 @@
+import { describe, expect, it } from 'vitest';
+import { SubmitButton } from './submit-button.component';
+import { render, screen } from '~/test/utils';
+import { GBIcon } from '../../i18n/icons';
+describe('Submit Button', () => {
+  it('should render a submit button with text only', () => {
+    render(<SubmitButton isDisabled>Test text</SubmitButton>);
+    expect(screen.getByRole('button', { name: 'Test text' })).toBeInTheDocument();
+  });
+  it('should render a submit button with an icon only', () => {
+    render(
+      <SubmitButton isDisabled>
+        <GBIcon />
+      </SubmitButton>,
+    );
+    expect(screen.getByRole('button', { name: 'United kingdom Flag' })).toBeInTheDocument();
+  });
+  it('should render a submit button with text and an icon', () => {
+    render(
+      <SubmitButton isDisabled>
+        <span>
+          <GBIcon />
+          <p>Test text</p>
+        </span>
+      </SubmitButton>,
+    );
+    expect(screen.getByRole('button', { name: 'United kingdom Flag Test text' })).toBeInTheDocument();
+  });
+  it('should render a submit button in the disabled state', () => {
+    render(<SubmitButton isDisabled>Test text</SubmitButton>);
+    expect(screen.getByRole('button', { name: 'Test text' })).toBeDisabled();
+  });
+  it('should render a submit button not in the disabled state', () => {
+    render(<SubmitButton isDisabled={false}>Test text</SubmitButton>);
+    expect(screen.getByRole('button', { name: 'Test text' })).toBeEnabled();
```

Comment:
Don't use negation to test, this is more an English this, make the test `should render a submit button in the enabled state`. You don't need to pass `isDisabled` if it is to be enabled, in fact, looking at the component, I see this is a mandatory attribute, which it shouldn't be. See if you can make `isDisabled` be optional.